### ISSUE TITLE
Paper: DIN-SQL: Decomposed In-Context Learning of Text-to-SQL with Self-Correction

### ISSUE THREAD

**Issue by konabuta** [konabuta/paper-survey#11]
### Summary by Copilot

- **DIN-SQL** stands for **Decomposed In-Context Learning of Text-to-SQL with Self-Correction**. [It is a method that breaks down the text-to-SQL task into multiple steps and feeds the intermediate results to LLMs to improve their performance and accuracy](https://arxiv.org/abs/2304.11015)¹².
- **Text-to-SQL** is a challenging task that aims to translate natural language utterances into SQL queries that can be executed on a relational database. For example, given the utterance "find employees who make more than their managers" and the schema of tables, one may want to generate a query in SQL that retrieves those employees from a database¹².
- **LLMs** are pre-trained neural models that can generate natural language and code based on a given context or prompt. Examples of LLMs are GPT-4, CodeX, and BART¹². LLMs can perform text-to-SQL tasks using zero-shot or few-shot prompting, which means they do not require any fine-tuning or additional training data¹².
- **Self-Correction** is a technique that allows LLMs to correct their own mistakes by comparing the generated SQL query with the natural language utterance and the database schema. If there is a mismatch, the LLM can revise the query until it matches the intended meaning¹².
- The paper claims that DIN-SQL outperforms existing methods on two text-to-SQL benchmarks: Spider and BIRD. On the Spider dataset, DIN-SQL achieves an execution accuracy of 85.3%, surpassing the previous state-of-the-art of 79.9%. On the BIRD dataset, DIN-SQL achieves an execution accuracy of 55.9%, setting a new state-of-the-art¹².
- The paper was submitted to NeurIPS 2023, a prestigious conference on neural information processing systems¹. The paper is available on arXiv¹ and GitHub³.

**Summary:** DIN-SQL is a novel method for generating SQL queries from natural language using large language models and self-correction. It achieves state-of-the-art results on two text-to-SQL benchmarks. The paper is submitted to NeurIPS 2023 and can be accessed online.

<details><summary>reference</summary>
Source: Conversation with Bing, 2/28/2024
([1](https://arxiv.org/abs/2304.11015)) DIN-SQL: Decomposed In-Context Learning of Text-to-SQL .... https://arxiv.org/abs/[2](https://ar5iv.labs.arxiv.org/html/2304.11015)304.11015.
(2) DIN-SQL: Decomposed In-Context Learning of Text-to-SQL .... https://ar5iv.labs.arxiv.org/html/2304.[1](https://arxiv.org/abs/2304.11015)1015.
(3) GitHub - madhup-google/DIN-SQL: https://arxiv.org/pdf/[2](https://ar5iv.labs.arxiv.org/html/2304.11015)304. .... https://github.com/madhup-google/DIN-SQL.
(4) DIN-SQL: Decomposed In-Context Learning of Text-to-SQL .... https://deepai.org/publication/din-sql-decomposed-in-context-learning-of-text-to-sql-with-self-correction.
(5) undefined. https://doi.org/[1](https://arxiv.org/abs/2304.11015)0.48550/arXiv.[2](https://ar5iv.labs.arxiv.org/html/2304.11015)304.[1](https://arxiv.org/abs/2304.11015)1015.
(6) undefined. https://arxiv.org/pdf/[2](https://ar5iv.labs.arxiv.org/html/2304.11015)304.
(7) undefined. https://arxiv.org/pdf/2304.[1](https://arxiv.org/abs/2304.11015)[1](https://arxiv.org/abs/2304.11015)0[1](https://arxiv.org/abs/2304.11015)5.pdf.

</details>

**Comment by konabuta** [comment]
## Spider dataset
- Use two official evaluation metrics
   - execution accuracy (EX)
   - exact set match accuracy (EM)
- Models: Codex family (Davinci and Cushman), and GPT-4 

<img width="429" alt="image" src="https://github.com/konabuta/my-survey/assets/3815677/e1a5a049-7390-4287-830a-95ef1c39b5c3">

**Comment by konabuta** [comment]
## Bird Benchmark
- introduced valid efficiecy score (VES)
<img width="864" alt="image" src="https://github.com/konabuta/my-survey/assets/3815677/503fc6fe-c198-4291-a3ec-603ef020d6eb">

**Comment by konabuta** [comment]
## Methodology
1. Schema Linking
2. Query Classification and decomposition
3. SQL generation
4. Self-correction

<img width="872" alt="image" src="https://github.com/konabuta/my-survey/assets/3815677/b6e5dc06-9888-46e5-a6d7-af4916cf4b43">

**Comment by konabuta** [comment]
## Schema Linking Module
- Identifying references to **database schema** and **condition values** in natural language queries.
<img width="529" alt="image" src="https://github.com/konabuta/my-survey/assets/3815677/dc9298bf-1796-498b-98e5-5adc5ecfd744">

**Comment by konabuta** [comment]
## Classification & Decomposition Module
- Challenge: Select correct table to be joined.
- Classify each query into "easy", "non-nested complex", and "nested complex".
- Detect the set of tables to b e joined.

<img width="529" alt="image" src="https://github.com/konabuta/my-survey/assets/3815677/d678274b-4890-4b88-badd-ad532fccda50">

**Comment by konabuta** [comment]
## SQL Generation Module
- easy class
   - a simple few-shot prompting with no intermediate steps
- non-nested complex class (require join)
   - use NatSQL

## Self-correction Module
- Challenge: missing or redundant keywords (DESC, DISTINCT, and agg func)
- generic
   - identify and correct the error in buggy sql (generated sql?)
- gentle
   - not assume the sql query is buggy.
   - ask model to check for any potential issues and provides some hints on the clauses to be checked.
